src/library/crate/crateschema.h

```diff
@@ -0,0 +1,27 @@
+#ifndef MIXXX_CRATESCHEMA_H
+#define MIXXX_CRATESCHEMA_H
+
+
+#include <QString>
+
+
+#define CRATE_TABLE "crates"
+#define CRATE_TRACKS_TABLE "crate_tracks"
+
+const QString CRATETABLE_ID = "id";
+
+// TODO(XXX): Fix database design flaw.
```

Comment:
> Sure, I don't want to touch this now! AutoDJ needs a new design before
extending it substantially.

> Performing a join or subselect should be ok and could be encapsulated into
additional views that are created and maintained by AutoDJ. I could even
imagine to separate AutoDJ completely from the main database and use an API
for interfacing with the library. Joins should be used sparely, because they
introduce a very strong coupling. AutoDJ is more of a high level workflow
concept, very different from managing the core library with tracks, crates
and playlists.

OK, Please add into the comment that a design doc is required before touch this issue.